apex/contrib/csrc/optimizers/multi_tensor_distopt_adam_kernel.cu

```diff
@@ -194,6 +194,175 @@ struct DistAdamFunctor
+  {
+    assert(noop_gmem);
+    assert(grad_scale_ptr);
+    assert(step);
+    assert(lr);
+
+    if(*noop_gmem == 1)
+      return;
+
+    float beta1_correction = 1.0f, beta2_correction = 1.0f;
+    if (bias_correction == 1) {
+      beta1_correction = 1 - pow(beta1, *step);
+      beta2_correction = 1 - pow(beta2, *step);
+    }
+
+    int tensor_loc = tl.block_to_tensor[blockIdx.x];
+    int chunk_idx = tl.block_to_chunk[blockIdx.x];
+    int n = tl.sizes[tensor_loc];
+
+    const float grad_scale = *grad_scale_ptr;
+
+    T* p_in = (T *)tl.addresses[0][tensor_loc];
+    p_in += chunk_idx*chunk_size;
+    T* m = (T *)tl.addresses[1][tensor_loc];
+    m += chunk_idx*chunk_size;
+    T* v = (T *)tl.addresses[2][tensor_loc];
+    v += chunk_idx*chunk_size;
+    const GRAD_T* g = (GRAD_T *)tl.addresses[3][tensor_loc];
+    g += chunk_idx*chunk_size;
+    PARAM_OUT_T* p_out = (PARAM_OUT_T *)tl.addresses[4][tensor_loc];
+    p_out += chunk_idx*chunk_size;
+
+    n -= chunk_idx*chunk_size;
+    n = chunk_size < n ? chunk_size : n;
+
+    const bool aligned = (n % ILP == 0 &&
+                          is_aligned(p_in) &&
+                          is_aligned(m) &&
+                          is_aligned(v) &&
+                          is_aligned(g) &&
+                          is_aligned(p_out));
+
+    for (int i_start = threadIdx.x*ILP; i_start < n; i_start += blockDim.x*ILP) {
+      T local_p[ILP];
+      T local_m[ILP];
+      T local_v[ILP];
+      GRAD_T local_g[ILP];
+      PARAM_OUT_T local_p_out[ILP];
+
+      // Load
+      if (aligned) {
+        load_store(local_p, p_in + i_start);
+        load_store(local_m, m + i_start);
+        load_store(local_v, v + i_start);
+        load_store(local_g, g + i_start);
+      } else {
+#pragma unroll
+        for (int ii = 0, i = i_start; ii < ILP; ii++, i++) {
+          if (i < n) {
+            local_p[ii] = p_in[i];
+            local_m[ii] = m[i];
+            local_v[ii] = v[i];
+            local_g[ii] = g[i];
+          } else {
+            local_p[ii] = 0;
+            local_m[ii] = 0;
+            local_v[ii] = 0;
+            local_g[ii] = 0;
+          }
+        }
+      }
+
+      // Local compute
+      local_step(
+        local_p, local_m, local_v, local_g, grad_scale,
+        beta1, beta2, beta1_correction, beta2_correction,
+        eps, *lr, mode, weight_decay);
+#pragma unroll
+      for (int ii = 0; ii < ILP; ii++) {
+        local_p_out[ii] = static_cast<PARAM_OUT_T>(local_p[ii]);
+      }
+
+      // Store
+      if (aligned) {
+        load_store(p_in + i_start, local_p);
+        load_store(m + i_start, local_m);
+        load_store(v + i_start, local_v);
+        load_store(p_out + i_start, local_p_out);
+      } else {
+#pragma unroll
+        for (int ii = 0, i = i_start; ii < ILP; ii++, i++) {
+          if (i < n) {
+            p_in[i] = local_p[ii];
+            m[i] = local_m[ii];
+            v[i] = local_v[ii];
+	    p_out[i] = local_p_out[ii];
```

Comment:
could you use the same indent as the lines above?

Reply:
Done. Formatted the code with clang-format Google style.